### ISSUE TITLE
Change robot model

### ISSUE THREAD

**Issue by Ethan0207** [reiniscimurs/DRL-robot-navigation#107]
Hi,
These days, I would like to replace the robot model with my own four-wheel differential model. But I don't know how to replace it. Could you give me some suggestions? Where should I change?

**Comment by reiniscimurs** [comment]
Hi,

You can launch your own robot from the launch file here: https://github.com/reiniscimurs/DRL-robot-navigation/blob/main/TD3/assets/multi_robot_scenario.launch#L7-L11

Please see other issues on changing the robot model, like: https://github.com/reiniscimurs/DRL-robot-navigation/issues/82

**Comment by Ethan0207** [comment]
Hi,
Thank you for your reply. I have changed my robot, but I meet some difficulties. When I use command " python3 train_velodyne_td3.py", in rviz, there is only the model. Other than that, nothing happened. And the terminal displays  errors.
![1 24](https://github.com/reiniscimurs/DRL-robot-navigation/assets/138771150/d0038b7f-d6fa-4b55-8218-a611e2eab2a7)
![1 24 2](https://github.com/reiniscimurs/DRL-robot-navigation/assets/138771150/890ca269-01d7-4539-9e63-4f568135e452)
This is my multi_robot_scenario.launch file.
![1 24 3](https://github.com/reiniscimurs/DRL-robot-navigation/assets/138771150/3283c56e-e14f-4afd-95ae-490c86764d2b)
This is my mini.gazebo.launch file.
![1 24 4](https://github.com/reiniscimurs/DRL-robot-navigation/assets/138771150/4d74cf9b-0be2-4817-a89f-fbe57b9427fd)
This is my xacro file.
![1 24 5](https://github.com/reiniscimurs/DRL-robot-navigation/assets/138771150/9d524680-d041-41b7-bd15-142b19d9125e)
![1 24 6](https://github.com/reiniscimurs/DRL-robot-navigation/assets/138771150/55e6002f-b894-4294-b7c2-9eba19a97564)
![1 24 7](https://github.com/reiniscimurs/DRL-robot-navigation/assets/138771150/a996077c-625f-4323-ad45-104a044cb427)
And in the velodyne_env.py, I just modified the vehicle name
![1 24 8](https://github.com/reiniscimurs/DRL-robot-navigation/assets/138771150/6de14f62-e4de-4fcd-9c92-bde6b0a22172)
![1 24 9](https://github.com/reiniscimurs/DRL-robot-navigation/assets/138771150/74936c22-273f-4945-a89b-75e8ea4100c2)
So could you give me some suggestions?

**Comment by reiniscimurs** [comment]
Hi,

in the first image you can see that you have an error. So it seems your subscriber is not getting any odom information from your subscriber. You should check your topic names and if they are subscribed to what they should be. Also, the issue might be in differential drive plugin so you should check that for your robot

**Comment by Ethan0207** [comment]
Hi，
Thank you for your reply. Today,  I checked all the nodes and topics, and it seemed normal. But it don’t send "/cmd_vel" topics.  And I checked my differential drive plugin. It seems to be fine. I don't konw how to solve it. Could you give me some suggestions?
Thank you very much.
This is my differential drive plugin file.
![1 26](https://github.com/reiniscimurs/DRL-robot-navigation/assets/138771150/e9a96902-fe35-4717-9bae-90b37a43c649)
![1 26-2](https://github.com/reiniscimurs/DRL-robot-navigation/assets/138771150/12ea4603-7306-4e6b-bf08-eedfda7a9227)

**Comment by Ethan0207** [comment]
Added:
When I first use command " python3 train_velodyne_td3.py", it prints many errors. And when I kill it, then use command " python3 train_velodyne_td3.py" again, it shows differently, but it still has errors.
![1 26 3](https://github.com/reiniscimurs/DRL-robot-navigation/assets/138771150/2d778352-2f67-4e16-9073-a9663866919e)
![1 26 4](https://github.com/reiniscimurs/DRL-robot-navigation/assets/138771150/065390c6-941b-4c4c-b5a3-abe655bde1cc)

**Comment by reiniscimurs** [comment]
I would suggest to first try understanding the error messages yourself, backtracking them and debugging them. And only after you have sincerely tried to understand (and googled them) to ask about issues here. There are many google answers to your p gain errors (which are not really errors). You are asking questions about things that are not part of this repo and something that you should be working on your own end about your own changes. Do actually investigate on your own end before asking questions.


For the other issue, your odom value is None because your subscriber is not returning any values. As you can see from your topics in rostopic list, your plugin does not seem to use robot namespace. So your topic names do no match where you subscribe to them and where they are published.

**Comment by Ethan0207** [comment]
Hi，
Thank you for your patient response. I had solved my problems. The problem was the namespace. However, from topic list and rqt, “goal_point”"linear_velocity""angular_velocity" didn't publish. From rviz, we can't see the target point and the movement path. What could be the reason for this? And in the code, where are the set target points and action tracks displayed?
![1 30 1](https://github.com/reiniscimurs/DRL-robot-navigation/assets/138771150/ad07a125-06fb-4bb1-aadf-de8d654a3000)
![1 30 2](https://github.com/reiniscimurs/DRL-robot-navigation/assets/138771150/37282b11-e865-4638-8aa7-c5869057cfc3)

**Comment by reiniscimurs** [comment]
Are you sure they are not published? Because they are in your rostopic list. If you want to check if anything is published to a topic you can just read it at runtime with `rostopic echo your_topic_name`. If it outputs anything, it is eing published.

It seems to me that you are just not displaying anything in rviz. I suggest to familiarize yourself with rviz and follow some tutorials on how to display information there.

**Comment by Ethan0207** [comment]
Thank you for your reply. I will try it.
